test/e2e/apps/deployment.go

```diff
@@ -880,12 +884,19 @@ func testProportionalScalingDeployment(f *framework.Framework) {
 	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
 	Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Verifying that first rollout's replicaset has 11 available replicas")
-	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 11)
+	numReplicaDiff := newReplicas - replicas
+	if numReplicaDiff < 0 {
+		numReplicaDiff = -numReplicaDiff
+	}
+	biggerProportion := (numReplicaDiff + 1) / 2
+	availableReplicas = availableReplicas + biggerProportion
```

Comment:
According to documentation: `With proportional scaling, we spread the additional replicas across all ReplicaSets. Bigger proportions go to the ReplicaSets with the most replicas and lower proportions go to ReplicaSets with less replicas. Any leftovers are added to the ReplicaSet with the most replicas. ReplicaSets with zero replicas are not scaled up.`

To clarify the formula, we have 2 ReplicaSets, thus `/ 2`. Note that `+ 1` is necessary to round up for bigger proportion.

Reply:
This is not right. This just splits the diff equally to both RSes, and give the rest to the larger RS. RSes should be scaled proportionally to their original fraction of replicas. 

We should probably change the maxSurge/maxUnavailable/replicas setting to make sure this (simply splitting the diff) does not pass.

Reply:
> splits the diff equally to both RSes, and give the rest to the larger RS
It is intended to _only_ work with no. of replicaset = 2.

The algorithm behind proportional scaling is non-trivial, and should have been unit tested thoroughly. IMO, this e2e test should emphasize on ensuring proportional scaling does work in a real-world scenario, but not about correctness of the algorithm behind (which should be unit tested heavily instead).

In fact, I think it is totally fine to just hard-code the values for this test and point it to documentation. Add a note to change the test if the documentation changes. This test should not be generalized with existence of unit tests.

Reply:
The unit tests can be found here: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/sync_test.go#L87-L201

In general, we should prevent hard-coding values for e2e tests. However, for this test, we should not do so as it is:
1. partially reinvent the wheel
2. make the test harder to understand (thus maintain) in future